sdk/src/wallet/utxo.ts

```diff
@@ -17,6 +17,76 @@ export const getBtcNetwork = (name: BitcoinNetworkName) => {
 
 type Output = { address: string; amount: bigint } | { script: Uint8Array; amount: bigint };
 
+class TreeNode<T> {
+    val: T;
+    children: TreeNode<T>[];
+
+    constructor(val: T, children: TreeNode<T>[] = []) {
+        this.val = val;
+        this.children = children;
+    }
+}
+
+const createUtxoNodes = (utxos: UTXO[], cardinalOutputsSet: Set<string>) =>
+    utxos.reduce(
+        (acc, utxo) => {
+            if (!cardinalOutputsSet.has(OutPoint.toString(utxo)))
+                // mark node as containing ordinals
+                acc.push(new TreeNode<OutputNodeData>({ ...utxo, cardinal: false }));
+            else acc.push(null);
+
+            return acc;
+        },
+        [] as (TreeNode<OutputNodeData> | null)[]
+    );
+
+const processNodes = async (
+    rootNodes: (TreeNode<OutputNodeData> | null)[],
+    cardinalOutputsSet: Set<string>,
+    esploraClient: EsploraClient,
+    ordinalsClient: OrdinalsClient
+) => {
+    const queue = Array.from(rootNodes);
+
+    while (queue.length > 0) {
+        const childNode = queue.shift();
+
+        if (childNode === null) continue;
+
+        const transaction = await esploraClient.getTransaction(childNode.val.txid);
+
+        if (transaction.status.confirmed) {
+            // if confirmed check if it contains ordinals
+            childNode.val.cardinal = cardinalOutputsSet.has(OutPoint.toString(childNode.val));
+        } else {
+            const response = await ordinalsClient.getInscriptionsFromOutPoint(childNode.val);
```

Comment:
What happens if the utxo is unconfirmed? Would the next line panic?

Reply:
it won't
```bash
curl -s -H "Accept: application/json" "https://ordinals-mainnet.gobob.xyz/output/9cd84b0d89695c732f42906e755894319f6afa1aa69fef0b9a0bc5974eed8ef9:2" 
```